### ISSUE TITLE
question on modeltime workflow - Why Calibrate with `modeltime_calibrate()`? 

### ISSUE THREAD

**Issue by ssh352** [business-science/modeltime#92]
I apologize if this is not the correct place to ask, but could someone help explain

https://cran.r-project.org/web/packages/modeltime/vignettes/getting-started-with-modeltime.html
 Step 4 - Calibrate the model to a testing set.? and modeltime_calibrate()

I've read Rob Hyndman's time series forecast book https://otexts.com/fpp3 but I can't find any pages related to model calibration on test set.

**Comment by mdancho84** [comment]
@ssh352 Thanks for reaching out. 

## Why Calibrate?

> Calibration uses out-of-sample error as a measure of future confidence

**The Calibrate step is a best practice borrowed from machine learning.** Normally we take a sample of the data, split into a train/test set. Build a model on the train set. Test on the test set (understand confidence). We then refit on the entire dataset (both train/test) and forecast forward. The confidence interval from the calibrate (train/test) process follows, and this is how we get confidence intervals.  

## Traditional Forecasting Uses In-Sample

In forecasting in general, most tools don't adopt the machine learning approach and rather use an in-sample error to estimate the prediction error. Why? 

- In-sample has worked well enough in the past by applying statistical factors to expand the error regions
- Out-of-sample requires 2 models rather than in-sample which is only 1. 

## Why Calibrate, Round 2?

The out-of-sample error is most important to me because it better approximates the true confidence of the model. I get worried when in-sample metrics are used to estimate error and confidence. I feel more confident when I use out-of-sample to measure confidence. 

So that's why I do it and I've added it to the workflow. You can by-pass this by skipping the `modeltime_calibrate()` step and jumping straight to forecasting:

- Advantage is this only creates 1 model
- Disadvantage is you don't get a confidence interval.

**Comment by mdancho84** [comment]
Closing this issue since it's not actually an issue but a question. We can keep the dialogue going if needed.

**Comment by ssh352** [comment]
@mdancho84  Thanks for the explanation. I had the question because I read something in Max Kuhn's book and I am still not 100% sure which side is correct.

Max Kuhn @topepo  the author of tidymodels, writes here: 

https://bookdown.org/max/FES/data-splitting.html

"
 The training set is used to develop models and feature sets; they are the substrate for estimating parameters, comparing models, and all of the other activities required to reach a final model. The test set is used only at the conclusion of these activities for estimating a final, unbiased assessment of the model’s performance. It is critical that the test set not be used prior to this point. Looking at the test sets results would bias the outcomes since the testing data will have become part of the model development process."

He thinks 
- the test set is only used for final model assessment. 
- model comparison is done without test set.
- test set is only used for final model assessment. Can't look at test set and go back to model comparison or selection. 

In modeltime workflow,  modeltime_calibrate() uses test set, and then modeltime_forecast() forecasts on test set. What do you think about the quoted writing above?

regarding confidence interval, take linear regression for example, here I can get the confidence interval and prediction interval without looking at the test set before predict(). So why do you think that you need look at test set to get the confidence interval? Thanks again.

```
attach(faithful)     # attach the data frame
  eruption.lm = lm(eruptions ~ waiting)
  newdata = data.frame(waiting=80)
  p1 <- predict(eruption.lm, newdata, interval="confidence")
  # prediction interval much wider
  p2 <- predict(eruption.lm, newdata, interval="predict")
  detach(faithful)     # clean up
```

**Comment by mdancho84** [comment]
Max has good comments. He is a very talented and experienced data scientist. I own several of his books. Applied Machine Learning is one of my favorites. And I agree with him. 

Using the test set to derive confidence is what I have determined the best method for my confidence in the model. My preference is to use out of sample data. It's up to you to decide if it's the method that is right for your needs. 

Regarding prediction interval, if you look into this it's alarming to me because it's derived from statistical measures on insample model error. This is not good if for example you have a model that is severely overfitting. Your prediction interval will be great but your forecast will be much worse (and outside your prediction interval). 

So this is why I prefer my method of calibration on a test set.
